src/form.ts

```diff
@@ -98,36 +105,92 @@ export class Form {
         this.openWidget();
       }
     });
+
+    return formCollapsed;
   }
 
   /**
    * Create close button
    *
+   * @param onClick - Actions on close click
    * @returns {HTMLButtonElement}
    */
-  private createClose(): HTMLButtonElement {
-    const closeContainer = make('button', classes.close) as HTMLButtonElement;
+  private createClose(onClick?: () => void): HTMLButtonElement {
+    const closeContainer = make('button', classes.close, {
+      type: 'button',
+    }) as HTMLButtonElement;
 
     closeContainer.innerHTML = IconCross;
 
-    closeContainer.addEventListener('click', () => {
-      this.collapseWidget();
+    closeContainer.addEventListener('click', (e) => {
+      e.stopPropagation();
+      onClick?.();
     });
 
     return closeContainer;
   }
 
+  /**
+   * Create notification after submit
+   *
+   * @param submitPromise - Promise created by submit
+   */
+  private createSubmitNotification(submitPromise: Promise<void>): void {
+    this.formCollapsed.replaceChildren();
+    const submission = this.fullFormConfiguration.submission;
+    const notificationContainer = make('span', classes.notification);
+
+    notificationContainer.innerHTML = IconLoader;
+
+    this.formCollapsed.append(notificationContainer);
+
+    const autoClose = (): void => {
+      this.container.classList.add(classes.hidden);
+    };
+
+    submitPromise
```

Comment:
Please use async/await syntax, it's much readable

Reply:
I'm not sure. Right now we do finally before then/catch. 
It we use async/await, for catching we change to try/catch and finally have to be added to both block, right?
 :thinking:
 
```ts
    try {
      await submitPromise;
      notificationContainer.appendChild(
        make('div', classes.successIcon, {
          textContent: '🎉',
        })
      );
    
      notificationContainer.appendChild(
        make('div', classes.notificationContent, {
          textContent: submission?.success || '',
        })
      );
    } catch {
      notificationContainer.appendChild(
        make('div', classes.notificationContent, {
          textContent: submission?.error || '',
        })
      );
    }

```